addons/l10n_in/models/account_invoice.py

```diff
@@ -92,7 +92,7 @@ def _post(self, soft=True):
                     move.l10n_in_state_id = self._l10n_in_get_indian_state(move.partner_id)
                 #still state is not set then assumed that transaction is local like PoS so set state of company unit
                 if not move.l10n_in_state_id:
-                    move.l10n_in_state_id = company_unit_partner.state_id
+                    move.l10n_in_state_id = company.state_id
```

Comment:
It is weird to still keep the l10n_in_state_id if the only reason is the l10n_in.state_in_ot

Reply:
@jco-odoo l10n_in_state_id is not for this only but this is Place of Supply and there is another PR of this where we not remove it but simplify it.